### ISSUE TITLE
'zoxide init --cmd cd bash' causing using cd to crash terminal

### ISSUE THREAD

**Issue by xyzme2** [ajeetdsouza/zoxide#694]
Hello,

just installed zoxide and I seem to be having difficulties with the --cmd flag. If i set the line in my bashrc to `eval "$(zoxide init --cmd cd bash)"` then when I use cd it will hang for a minuite or so then either do nothing or crash the terminal. 
 
Interestingly it only seems to happen when it is set to cd. Though setting it to 'ls' gives the following error message when I boot my terminal
```
bash: eval: line 130: syntax error near unexpected token `('
bash: eval: line 130: `ls() {'
```
setting it to other options seems to work fine (i.e. `j`, `test`) it seems for now a simple alias will work, but I'm certain the --cmd flag exist for a reason. 

Am I missing something on how to set up or is this non-expected behavior?

**Comment by MaximSrour** [comment]
I also have this issue in some cases.

On Windows 10 through WSL (Ubuntu 22.04.2) the commands work just fine. On a native Linux machine (Ubuntu Server 22.04.3) it has the same behaviour. Rebinding z and zi to cd and cdi will cause the crash, as well as leaving the z and zi commands as is.

I am using SSH from my WSL shell into the native box.

**Comment by M0hammedImran** [comment]
I am facing the same issue in zsh.
This is in my `.zshrc`
![image](https://github.com/ajeetdsouza/zoxide/assets/52968617/cd4b3223-b734-44fc-8f93-2b7b791ea655)

When I run `cd` I get this error
![image](https://github.com/ajeetdsouza/zoxide/assets/52968617/bf3d276e-8dee-437e-95ed-984af493e324).

![image](https://github.com/ajeetdsouza/zoxide/assets/52968617/05b98f21-e809-4adc-b356-70390096f9fd)

**Comment by didi-maru** [comment]
I have a similar issue with fish

**Comment by TheBloodyScreen** [comment]
I have the same issue using both bash and zsh.

**Comment by NoNamePaul94** [comment]
I also have this issue with ZSH 
``maximum nested function level reached; increase FUNCNEST?``

**Comment by favna** [comment]
Chiming in that I'm facing the same issue with zsh, but only on my Linux servers, one is running Ubuntu (server) 22.04.3 LTS x86_64 and the other Debian GNU/Linux 11 (bullseye) x86_64. On my host (macOS 14.2.1 23C71 x86_64) it works just fine. All of them use similar setups of zsh + oh-my-zsh as shell and [Starship](https://starship.rs) as prompt for as much as that is worth. I would apply the workaround from NoNamePaul, but even delegating it to a separate file to source, it's still quite a bit of stuff for what should be an easy setup.

**Comment by HarryPoggers** [comment]
+1 for me as well on Ubuntu 22.04 WSL running with `eval "$(zoxide init --cmd cd bash)"`

**Comment by 0PandaDEV** [comment]
same issue for me here cd also returns 
`cd:1: maximum nested function level reached; increase FUNCNEST?`

**Comment by ajeetdsouza** [comment]
Debian / Ubuntu derivatives package a very old version of zoxide in their repositories. The latest version shouldn't have this problem - could you try using the install script?

```sh
curl -sS https://raw.githubusercontent.com/ajeetdsouza/zoxide/main/install.sh | bash
```

**Comment by M0hammedImran** [comment]
> Debian / Ubuntu derivatives package a very old version of zoxide in their repositories. The latest version shouldn't have this problem - could you try using the install script?
> 
> ```shell
> curl -sS https://raw.githubusercontent.com/ajeetdsouza/zoxide/main/install.sh | bash
> ```

I reinstalled using `cargo`.
![image](https://github.com/ajeetdsouza/zoxide/assets/52968617/70101e5a-de71-46a0-851f-d94d014a42a2)

It works.

**Comment by wlarch** [comment]
I had the same issue as mentioned by others when installing using Debian `apt` package manager.
Installing `zoxide` using the installation script did the trick :

`curl -sS https://raw.githubusercontent.com/ajeetdsouza/zoxide/main/install.sh | bash`

**Comment by iskaron** [comment]
> Debian / Ubuntu derivatives package a very old version of zoxide in their repositories. The latest version shouldn't have this problem - could you try using the install script?
> 
> ```shell
> curl -sS https://raw.githubusercontent.com/ajeetdsouza/zoxide/main/install.sh | bash
> ```

ah sorry for the inconvenience, I would have never thought of that. Yes, that fixed it for me, thanks!

**Comment by ajeetdsouza** [comment]
I've edited the README to strike out outdated package repositories to prevent future confusion. Thanks!

**Comment by kbcmdba** [comment]
Just had the same issue on my Ubuntu 22.04.3 system where I installed zoxide using

`sudo apt install zoxide fzf`

**Comment by ajeetdsouza** [comment]
@kbcmdba the solution is posted in the thread.

**Comment by kbcmdba** [comment]
Thanks for that. I got it. Everything seems to be working fine.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by yiqiangjizhang** [comment]
> I had the same issue as mentioned by others when installing using Debian `apt` package manager : `cd:1: maximum nested function level reached; increase FUNCNEST?`
> 
> Installing `zoxide` using the installation script did the trick : `curl -sS https://raw.githubusercontent.com/ajeetdsouza/zoxide/main/install.sh | bash`

This fixed it for me as well in my WSL Ubuntu 22.04
